app/src/main/java/com/triviagenai/triviagen/trivia/presentation/TriviaQuestionViewModel.kt

```diff
@@ -70,22 +79,27 @@ class TriviaQuestionViewModel @Inject constructor(
 
     fun processIntent(intent: TriviaIntent) {
         when (intent) {
-            is TriviaIntent.SubmitAnswer -> viewModelScope.launch { submitAnswer(intent.selectedOptionIndex) }
+            is TriviaIntent.SubmitAnswer -> viewModelScope.launch { submitAnswer(intent.selectedOptionIndex, intent.navController) }
```

Comment:
I believe navigation should be tied to the UI and should not be executed from the ViewModel.

Could we use something like:
```    LaunchedEffect(triviaRound.questions.size, triviaRound.currentQuestionIndex, questionBlock.selectedAnswer) {
        if (triviaRound.questions.isNotEmpty() &&
            triviaRound.questions.size - 1 == triviaRound.currentQuestionIndex &&
            questionBlock.selectedAnswer != SelectedAnswerState.Unanswered) {
            navController.navigate(Route.ResultsRoute) {
                popUpTo(Route.MainMenuRoute) {
                    inclusive = true
                }
            }
        }
    }
```
    
    This could be within the `DisplayGameContent` composable.